### DISCUSSION TITLE
screwy naming template

### DISCUSSION THREAD

**Discussion by dougK44883** [jessedp/tablo-tools-electron#156] [Show and tell]
It seems I have 2 naming templates which result in a file and as the literal template.

```
{"type":"episode","label":"pbs1","slug":"pbs1","template":"{{episodePath}}/{{lowercase airing_details.channel.channel.network}}/{{pascalcase (strip \" on Masterpiece\" stripShowTitle)}}/{{pascalcase (strip \" on Masterpiece\" stripShowTitle)}}-{{lPad episode.number}}-{{stripTitle}}.{{EXT}}","_id":"7qihTzisyXsexjtk"}
{"type":"episode","label":"pbs1a","slug":"pbs1a","template":"{{episodePath}}/{{lowercase airing_details.channel.channel.network}}/{{pascalcase (strip \" on Masterpiece\" stripShowTitle)}}/{{pascalcase (strip \" on Masterpiece\" stripShowTitle)}}-{{lPad episode.number}}.{{EXT}}","_id":"GIhEk3IeKJyKVMBq"}
```
oddly both result in a file name `{{pascalcase (strip  on Masterpiece stripShowTitle)}}-{{lPad episode.number}}-{{stripTitle}}.{{EXT}}.mp4` & `{{pascalcase (strip  on Masterpiece stripShowTitle)}}-{{lPad episode.number}}.{{EXT}}.mp4`
Well, the directory path works out.  From the output log it show ffmpeg outputting to the file, it's not the entire template not getting parsed out.
`Output #0, mp4, to '/media/VideoLibrary/tablo/tbd/Wipeout/{{pascalcase (strip  on Masterpiece stripShowTitle)}}-{{lPad episode.number}}-{{stripTitle}}.{{EXT}}.mp4':`

In the export list it shows the correct filename, no sign or any error.

I have a template with similar compound handlebars which work fine:
```
{"type":"episode","label":"Net-Show-Show-se-Title","slug":"net-show-show-se-title","template":"{{episodePath}}/{{airing_details.channel.channel.call_sign_src}}/{{pascalcase (strip \"The \" stripShowTitle)}}/{{pascalcase (strip \"The \" stripShowTitle)}}-{{episodeOrDate}}-{{stripTitle}}.{{EXT}}","_id":"OW1YHEhT6ZkH7CSd"}
``` 
I haven't fully dug in.  I have tried the offending templates on a mix of different show, same result.  Haven't changed it since v0.2.4 and it's not a regular use one, so if/when it _broke_ is unclear.

Just posting in case something jumps out to anyone else.  Later I'll swap variables and mix things up.  Suggestions? or **reasonable** guesses.

**Comment by jessedp** [comment]
This was a code error fixed in #157 / v0.3.7. 
You would have had to run something like `DEBUG=tablo* /opt/TabloTools/tablo-tools` from the command line (for the deb) to see the error (and way too much other stuff). It *might* have been in the `main.log` or the export log for the show if you have the app debug turned on.

**Reply by dougK44883** [reply]
Well ok then.  Awesome you tracked it down just-like-that!  After I posted, the lPad and tried to dig up where I found all those documented.  I'd still like to find them, but now I supposedly don't have to be too concerned.
## again, a big Thanks

The `debug=*ablo*` also logs the verbose from ffmpeg.  I suppose `-loglevel warning` would eliminate too much.  More about logs, as noted: "This doesn't clean itself up"  in `~/.config/tablo-tools/logs/`, sometimes an indivual log file is only 7 lines long while another logs events found in `main.log`.

Though not mentioned, the file `~/.config/tablo-tools/export-log.db`(actually a json file not a db) has grown to over 23MBs ...really need everything for months?

**Reply by jessedp** [reply]
The helper function docs are [here](http://jonschlinkert.github.io/template-helpers/#docs), though there are a ton of functions that make no sense in this context. Most of the relevant/useful string functions start at [lowercase](http://jonschlinkert.github.io/template-helpers/#lowercase).


The `DEBUG=*ablo*` is matching anything labeling its debug messages that match `*ablo*`, so is pretty much that same as `tablo*`. The ffmpeg messages are there because I'm writing them out with a label that starts with `tablo-tools`. Run it with `DEBUG=*` to *really* get a feel for everything that will write debug messages (that I don't even care about).

`-loglevel warning` really shouldn't do anything and I couldn't find it documented for node or electron. Using it seemed to add a timer to tons of calls that just slowed things down even more.

`export-log.db` was intended to be used to show historical stats and an indicator in the UI that something had been exported, but I've not done anything with it.

**Reply by jessedp** [reply]
Oh. `lPad` is the only one not in that set of helpers - part of why I added #73 , though haven't touched it.

`lPad` works like this:

`{{lPad FIELD LEN CHAR}}`

`FIELD` == the data field you want to left pad, so `episode.number` in your example

`LEN` == the length to pad to. So you probably want `2` (01) or `3` (001). Not required, defaults to `1`, which just returns whatever FIELD passed in

`CHAR` == the character to pad with. Not required, defaults to `0`.

So in your example (after I fixed my broken code), you wanted something like this:
`{{lPad episode.number 2}}`

**Comment by dougK44883** [comment]
The `-loglevel warning` was in reference to `ffmpeg`.  What little I do know (sort of).  Instead of hundreds of pointless lines of output it limits to only "warning" or greater.  https://manpages.debian.org/bullseye/ffmpeg/ffmpeg.1.en.html#loglevel 

The functions from the helpers are... super-wonderful.  I mostly like condensed names, `pascalcase` is more perfect than I never knew existed.  Another great one, getting rid of The from titles `strip "The "` and allow multiple/nested helper is just what my OCD needs.  The `lcase` and `lPad` are just convenient at times.

I appericate the lPad addition and the explanation.  

Thankfully I don't need to run through all the debug messages, but I'll recall them if there's a reason it may help.
